### PR TITLE
Added ability to disable oauth options, added some webapp tests

### PR DESCRIPTION
**Pull request by karenhanson**
- Added properties rmapweb.google-oauth-enabled, rmapweb.twitter-oauth-enabled, and rmapweb.orcid-oauth-enabled. These default to false and the system will check them in order to determine whether the corresponding OAuth buttons will appear. Where all 3 OAuth options are disabled the login link will disappear completely. Any attempt to reach a login page will redirect to the home page.  The properties that determine whether each oauth option is enabled affects
the layout of the page header on most views, so they are pulled in using a new SiteProperties class, and this object is added to the model on each page load using a HandlerInterceptor.postHandle().  This can be used for other site-wide properties in the future.  This closes issue #76.
- Added some tests for new SitePropertiesToModellnterceptor
- Previous controller tests were replicas of the DataDisplayServiceImplTests.  Removed most of these and added a simple HomeController test to be used as a possible model for other controller tests.
- DataDisplayServiceImplTests were not working, they were dependent on a live database - fixed these and added some additional tests.

### CODE REVIEW

File: webapp/src/test/java/info/rmapproject/webapp/WebDataRetrievalTestAbstract.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+import info.rmapproject.core.rdfhandler.impl.openrdf.RioRDFHandler;
+import info.rmapproject.core.rmapservice.RMapService;
+import info.rmapproject.core.rmapservice.impl.openrdf.triplestore.SesameSailMemoryTriplestore;
+import info.rmapproject.core.rmapservice.impl.openrdf.triplestore.SesameTriplestore;
+import info.rmapproject.testdata.service.TestConstants;
+import info.rmapproject.testdata.service.TestDataHandler;
+import info.rmapproject.testdata.service.TestFile;
+
+/**
+ * Tests for ResponseManager.
```
Comment (emetsger) [inline webapp/src/test/java/info/rmapproject/webapp/WebDataRetrievalTestAbstract.java]: Looks like a copy and paste?

File: webapp/src/test/java/info/rmapproject/webapp/WebDataRetrievalTestAbstract.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+	
+	/** The triplestore. */
+	@Autowired
+	protected SesameTriplestore triplestore;
+			
+	/** The context. */
+	protected ApplicationContext context;
+		
+	/**
+	 * Instantiates a new response manager test.
```
Comment (emetsger) [inline webapp/src/test/java/info/rmapproject/webapp/WebDataRetrievalTestAbstract.java]: Same here, a copy and paste comment (sorry being nit-picky)

File: webapp/src/test/java/info/rmapproject/webapp/controllers/SitePropertiesToModelInterceptorTest.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.springframework.web.servlet.ModelAndView;
+
+import info.rmapproject.webapp.WebTestAbstract;
+import info.rmapproject.webapp.utils.SiteProperties;
+
+/**
+ * Tests for SessionDefaultsInterceptor
```
Comment (emetsger) [inline webapp/src/test/java/info/rmapproject/webapp/controllers/SitePropertiesToModelInterceptorTest.java]: c & p

File: webapp/src/test/java/info/rmapproject/webapp/controllers/SitePropertiesToModelInterceptorTest.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+	        siteProperties = new SiteProperties(false, false, true);
+	        interceptor = new SitePropertiesToModelInterceptor(siteProperties);
+	        interceptor.postHandle(request, response, null, modelview);
+	        siteProps = (SiteProperties) modelview.getModel().get(SITE_PROPERTIES_ATTRIBNAME);
+	        assertFalse(siteProps.isGoogleEnabled());
+	        assertFalse(siteProps.isOrcidEnabled());
+	        assertTrue(siteProps.isTwitterEnabled());
+
+	        	        		        
+	    } catch (Exception e) {
```
Comment (emetsger) [inline webapp/src/test/java/info/rmapproject/webapp/controllers/SitePropertiesToModelInterceptorTest.java]: So these catch blocks aren't really necessary.

If something in the `try` block throws an exception, the exception can simply be declared in the method signature using a `throws` statement, and JUnit will handle the rest by failing the test and emitting the stacktrace (which will be present in the surefire report).  

So printing the stack trace and then calling fail() emulates the existing behavior of JUnit and would be considered redundant.  Removing the try/catch and the redundant exception handling makes the test a bit more readable!

File: webapp/src/test/resources/rmap.properties [hunk 5]
```diff
@@ -0,0 +1,24 @@
+#overridden test properties
+rmapweb.google-oauth-enabled=true
```
Comment (emetsger) [inline webapp/src/test/resources/rmap.properties]: Maybe update `integration/src/main/resources/rmap.properties` and remove any old auth properties and add sample (but commented out as appropriate) updated auth properties?

File: webapp/src/test/resources/servlet-test-context.xml [hunk 6]
```diff
@@ -1,73 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (emetsger) [inline webapp/src/test/resources/servlet-test-context.xml]: hooray

File: webapp/src/test/resources/rmapweb.properties [hunk 7]
```diff
@@ -1,36 +0,0 @@
-#Google OAUTH Config. Register for a key through a Google account at:
```
Comment (emetsger) [inline webapp/src/test/resources/rmapweb.properties]: hooray

File: webapp/src/test/resources/rmap.properties (hunk 5 above)
Comment (karenhanson) [reply webapp/src/test/resources/rmap.properties]: Added the properties but commented out.  They only need to be overridden if enabled.

File: webapp/src/test/java/info/rmapproject/webapp/WebDataRetrievalTestAbstract.java (hunk 2 above)
Comment (karenhanson) [reply webapp/src/test/java/info/rmapproject/webapp/WebDataRetrievalTestAbstract.java]: that's OK - I like nit-picky! 😃

**Pull request merged**